### ISSUE TITLE
giac does not support arbitrary precision floating point

### ISSUE THREAD

**Issue by tornaria** [void-linux/void-packages#28989]
### System

* xuname:  Void 5.4.99_1 x86_64 GenuineIntel uptodate FF
* package:  giac-1.5.0.87_3

### Expected behavior
Can choose arbitrary floating point precision:
```
$ giac
[...]
0>> Digits:=30
30
// Time 0
1>> evalf(pi)
3.14159265358979323846264338328
// Time 0
2>>
```

### Actual behavior
Floating point precision is capped at 13 (i.e. double precision, 53 bits)
```
$ giac
[...]
0>> Digits:=30
13
// Time 0
1>> evalf(pi)
3.14159265359
// Time 0
```

### Proposed fix

Add `libmpfr-devel` to `makedepends`.

**Comment by ericonr** [comment]
Hi! That sounds like a reasonable change! Would you like to make a PR for it?

**Comment by tornaria** [comment]
While doing that I run `xbps-src check` and found that several tests fail, revealing that other dependencies are missing:
- in the current version: 12 tests fail
- adding `mpfr-devel` to `makedepends`: 9 tests fail
- adding also `pari-devel`: 2 tests fail
- adding a patch found in [`sagemath`](https://trac.sagemath.org/ticket/25567#comment:41): 1 tests fail
I wasn't able to get rid of the last fail (a segmentation fault).

Should I do all of that in a single PR, or split?

Comments:
- For the remaining test failure, I will open another issue. I should point out that `sagemath` compiles giac (I think the same version as in void) and that one does NOT have the segfault so this might be worth looking into.
- There are more (optional?) dependencies for giac, some of them are available in void, namely: `libao-devel libsamplerate-devel libcurl-devel libICE-devel glpk-devel gmpxx-devel ecm-devel`. If adding them to `makedepends`, they are found by configure, but they don't affect tests. I'm not sure if I should also add those dependencies or not. In case of doubt, I'll stick to the changes described above (mpfr and pari)
- For the record, other (optional?) dependencies not available in void are `mpfi ntl CoCoA nauty`
- I'm not a direct user of `giac`, but I would like to package `sagemath` or at least make it easy to compile with as many system packages as possible. The original issue was caught by running sage tests.

**Comment by dkwo** [comment]
@tornaria I'm interested in helping with sage/giac, see also #19090

**Comment by ericonr** [comment]
Fixed in #29141
